services/uploads/serverless.yml

```diff
@@ -364,6 +382,94 @@ resources:
+      Type: AWS::EC2::SecurityGroup
+      Properties:
+        GroupDescription: Security group for ClamAV daemon
+        VpcId: ${self:custom.vpcId}
+        SecurityGroupIngress:
+          - IpProtocol: tcp
+            FromPort: 3310
+            ToPort: 3310
+            SourceSecurityGroupId: ${self:custom.sgId}
+
+    ClamAVInstanceProfile:
+      Type: AWS::IAM::InstanceProfile
+      Properties:
+        Path: '/delegatedadmin/developer/'
+        Roles:
+          - !Ref ClamAVInstanceRole
+
+    ClamAVInstanceRole:
+      Type: AWS::IAM::Role
+      Properties:
+        Path: '/delegatedadmin/developer/'
+        PermissionsBoundary: !Sub 'arn:aws:iam::${AWS::AccountId}:policy/cms-cloud-admin/ct-ado-poweruser-permissions-boundary-policy'
+        RoleName: !Sub 'clamavdVm-${sls:stage}-ServiceRole'
+        AssumeRolePolicyDocument:
+          Version: '2012-10-17'
+          Statement:
+            - Effect: Allow
+              Principal:
+                Service: ec2.amazonaws.com
+              Action: sts:AssumeRole
+        Policies:
+          - PolicyName: ClamAVInstancePolicy
+            PolicyDocument:
+              Version: '2012-10-17'
+              Statement:
+                - Effect: Allow
+                  Action:
+                    - logs:CreateLogGroup
+                    - logs:CreateLogStream
+                    - logs:PutLogEvents
+                  Resource: '*'
+
+    ClamAVInstance:
+      Type: AWS::EC2::Instance
+      Condition: IsDevValProd
+      Properties:
+        InstanceType: t3.medium
+        ImageId: ami-0c7217cdde317cfec # Ubuntu 22.04 LTS
+        IamInstanceProfile: !Ref ClamAVInstanceProfile
+        KeyName: mojo # for testing
+        NetworkInterfaces:
+          - AssociatePublicIpAddress: true
+            DeviceIndex: '0'
+            GroupSet:
+              - !Ref ClamAVSecurityGroup
+              - ${self:custom.sgId}
+            SubnetId: !Sub ${self:custom.publicSubnetA}
+        Tags:
+          - Key: Name
+            Value: clamavd-${sls:stage}
+          - Key: mcr-vmuse
+            Value: clamavd
+        UserData:
+          Fn::Base64: !Sub |
+            #!/bin/bash
+            apt-get update
+            apt-get install -y clamav clamav-daemon
+
+            echo '${self:custom.authorizedKeys}' > /home/ubuntu/.ssh/authorized_keys
+            chown ubuntu:ubuntu /home/ubuntu/.ssh/authorized_keys
+            chmod 600 /home/ubuntu/.ssh/authorized_keys
+
+            # Install clamav 1.0.4
+            wget https://www.clamav.net/downloads/production/clamav-1.0.4.linux.x86_64.deb
+            dpkg -i clamav-1.0.4.linux.x86_64.deb
+            apt-get install -f
+
+            # Write to the clamd.conf
+            echo "TCPSocket 3310" >> /etc/clamav/clamd.conf
+            echo "TCPAddr 0.0.0.0" >> /etc/clamav/clamd.conf 
+
+            # Start clamd and get defs
+            systemctl enable clamav-freshclam
+            systemctl start clamav-freshclam
```

Comment:
will Freshclam keep auto-updating the definitions in the background? That would be nice we could get rid of the fresh clam deamon?

Reply:
It does! It looks every hour, for example:

```
Mon Feb 12 16:00:19 2024 -> --------------------------------------
Mon Feb 12 17:00:19 2024 -> Received signal: wake up
Mon Feb 12 17:00:19 2024 -> ClamAV update process started at Mon Feb 12 17:00:19 2024
Mon Feb 12 17:00:19 2024 -> daily.cld database is up-to-date (version: 27183, sigs: 2053128, f-level: 90, builder: raynman)
Mon Feb 12 17:00:19 2024 -> main.cvd database is up-to-date (version: 62, sigs: 6647427, f-level: 90, builder: sigmgr)
Mon Feb 12 17:00:19 2024 -> bytecode.cvd database is up-to-date (version: 334, sigs: 91, f-level: 90, builder: anvilleg)
Mon Feb 12 17:00:19 2024 -> --------------------------------------
Mon Feb 12 18:00:19 2024 -> Received signal: wake up
Mon Feb 12 18:00:19 2024 -> ClamAV update process started at Mon Feb 12 18:00:19 2024
Mon Feb 12 18:00:19 2024 -> daily.cld database is up-to-date (version: 27183, sigs: 2053128, f-level: 90, builder: raynman)
Mon Feb 12 18:00:19 2024 -> main.cvd database is up-to-date (version: 62, sigs: 6647427, f-level: 90, builder: sigmgr)
Mon Feb 12 18:00:19 2024 -> bytecode.cvd database is up-to-date (version: 334, sigs: 91, f-level: 90, builder: anvilleg)
Mon Feb 12 18:00:19 2024 -> --------------------------------------
Mon Feb 12 19:00:19 2024 -> Received signal: wake up
Mon Feb 12 19:00:19 2024 -> ClamAV update process started at Mon Feb 12 19:00:19 2024
Mon Feb 12 19:00:19 2024 -> daily.cld database is up-to-date (version: 27183, sigs: 2053128, f-level: 90, builder: raynman)
Mon Feb 12 19:00:19 2024 -> main.cvd database is up-to-date (version: 62, sigs: 6647427, f-level: 90, builder: sigmgr)
Mon Feb 12 19:00:19 2024 -> bytecode.cvd database is up-to-date (version: 334, sigs: 91, f-level: 90, builder: anvilleg)
Mon Feb 12 19:00:19 2024 -> --------------------------------------
```

Which means yeah, we can refactor and drop the lambda that does the whole AV download to s3 process.

Reply:
omg I love this all so much